example/src/screens/Room.tsx

```diff
@@ -1,181 +1,187 @@
+import { BrandColors } from '@colors';
+import { BackgroundAnimation } from '@components/BackgroundAnimation';
 import { Icon } from '@components/Icon';
+import { RoomParticipant } from '@components/RoomParticipant';
+import { Typo } from '@components/Typo';
 import * as Membrane from '@jellyfish-dev/react-native-membrane-webrtc';
 import { RootStack } from '@model/NavigationTypes';
 import type { NativeStackScreenProps } from '@react-navigation/native-stack';
-import React, { useState, useEffect, useCallback } from 'react';
-import { StyleSheet, Text, View, Pressable } from 'react-native';
+import React, { useCallback, useEffect } from 'react';
+import { StyleSheet, View, Pressable, Dimensions } from 'react-native';
+import { useVideoroomState } from 'src/VideoroomContext';
 
-import { Controls } from '../Controls';
-import { Settings } from '../Settings';
+import { CallControls } from '../components/CallControls';
+
+const HEADER_AND_FOOTER_SIZE = 126;
+const PADDING_BETWEEN_PARTICIPANTS = 16;
+const MAX_NUM_OF_USERS_ON_THE_SCREEN = 8;
+const FLEX_BRAKPOINT = 3;
 
 type Props = NativeStackScreenProps<RootStack, 'Room'>;
 
 export const Room = ({ navigation }: Props) => {
+  const { width, height } = Dimensions.get('window');
+  const { roomName } = useVideoroomState();
   const participants = Membrane.useRoomParticipants();
-  const tracks = participants
-    .map((p) => p.tracks.filter((t) => t.type === 'Video'))
-    .flat();
-
-  const [focusedTrackId, setFocusedTrackId] = useState<string | null>(null);
-  const focusedTrack = tracks.find((t) => t.id === focusedTrackId);
-  const focusedParticipant = participants.find(
-    (p) => p.tracks.find((t) => t.id === focusedTrackId) != null
-  );
+
+  const videoViewWidth = (width - 3 * PADDING_BETWEEN_PARTICIPANTS) / 2;
+  const smallScreenVideoWidth =
+    (height -
+      HEADER_AND_FOOTER_SIZE -
+      PADDING_BETWEEN_PARTICIPANTS * (participants.length / 2 + 2)) /
+    Math.ceil(participants.length / 2);
 
   const { disconnect: mbDisconnect } = Membrane.useMembraneServer();
 
   useEffect(() => {
-    if (!focusedTrack && tracks[0]) {
-      setFocusedTrackId(tracks[0].id);
-    }
-  }, [tracks, focusedTrack]);
+    const handleBeforeRemoveEvent = (e) => {
```

Comment:
I do not understant why do we need to do all this handling. Can't we just set `gestureEnabled = false` on the screen?

Reply:
ok, add a comment with a link to this issue

Reply:
Added.